### DISCUSSION TITLE
Who is keeping pycharm test execution freezed? pytest, xdist or pycharm?

### DISCUSSION THREAD

**Discussion by andreabisello** [pytest-dev/pytest#11338] [Q&A]
Hi
i'm asking for a suggestion
sometime happen that my test exection with pycharm (everything updated) freeze in this way
![image](https://github.com/pytest-dev/pytest/assets/2149815/4ea2f4af-95f6-447d-b0a8-6e1a8409761b)
all the test are passed or ignored or failed,
108 on 108.
so who is waiting for who?
is pycharm? is xdist? is pytest?
any suggestion?
thanks.

**Comment by The-Compiler** [comment]
Your first step for this should be running your tests with `pytest` from the commandline. If it works there, it sounds like it's probably PyCharm.

**Reply by andreabisello** [reply]
Thanks @The-Compiler .
The problem is that happen rarely and without any reason.
I'm using pycharm+pytest+xdist every day in the last 4 years,
I do a lot of execution. but sometime something hangs.
Ok i will try from the command line skipping pycharm,
but this never happened to you?
thanks and have a good day

**Comment by andreabisello** [comment]
hi @The-Compiler 
i start to think the problem is xdist with pytest-timeout,
that crashes keeping pycharm not updated
you see that?
![image](https://github.com/user-attachments/assets/9a77f9fd-b0d6-46ba-ad35-b58202b465fe)
test_prepdv_cades has only one test inside created by the only parameter having prepdv_zip,
for pycharm the test is finished, but the "container of the test" have not finished.
running from local,
sometime i see errors like that one
![image](https://github.com/user-attachments/assets/afa3d037-83c3-4d51-84cd-59fc13ba0fe0)
a worker crash because of a timeout,
and nothing more happens.

now i'm gonna try to remove the pytest-timeout, 
and update you only for sharing knowledge, 
then i'm gonna open the bug in the right place.

thanks :)

**Reply by The-Compiler** [reply]
Looks to me like the PyCharm integration isn't handling that case correctly. I believe I've seen similar things when e.g. the test process segfaults (just no output in PyCharm), but personally I use VS Code so I might be misremembering.

**Reply by andreabisello** [reply]
thankyou, by the way can you list to me the plugins you use to use correcly pytest in vs code?

**Reply by The-Compiler** [reply]
I just use the default Python extension from Microsoft.
